### DISCUSSION TITLE
Valtio state async properties

### DISCUSSION THREAD

**Discussion by italodeandra** [pmndrs/valtio#91] [General]
If I understand correctly, when using async (like fetch on the example) data, it will be fetched as soon as the proxy is created (but it will be suspended until the data is ready).

Is it possible to only fetch after the data is requested for the first time? Because I have some remote data that will only be used on a specific page that is not open. And I want it to be fetched only if the user navigates to it.

**Comment by dai-shi** [comment]
Just put a promise afterward.
```js
const state = proxy({})

const userClicked = () => {
  state.result = fetch(url).then(res => res.json())
}
```

There's no notion of "data is requested" as it's just an object...

**Reply by italodeandra** [reply]
Yes, you're right. I'm learning proxies better before I can build a better question.
